interpreter/cling/lib/Interpreter/DefinitionShadower.cpp

```diff
@@ -149,14 +160,18 @@ namespace cling {
   ASTTransformer::Result DefinitionShadower::Transform(Decl* D) {
     Transaction *T = getTransaction();
     const CompilationOptions &CO = T->getCompilationOpts();
-    // Global declarations whose origin is the Cling prompt are subject to be
-    // nested in a `__cling_N5' namespace.
+
+    // For variable templates, Transform() is invoked with a VarDecl; get the
+    // corresponding VarTemplateDecl.
+    if (auto VD = dyn_cast<VarDecl>(D))
+      if (auto VTD = VD->getDescribedVarTemplate())
+        D = VTD;
+
+    // Disable definition shadowing for some specific cases.
     if (!CO.EnableShadowing
```

Comment:
Shouldn't we return early if `!CO.EnableShadowing`?

Reply:
True,  Fixing.